File: shell/shell.js
function shellSort(arr) {
    var len = arr.length,
        temp,
        gap = 1;
    while (gap < len/3) {
        gap = gap * 3 + 1;
    }
    for (gap; gap > 0; gap = Math.floor(gap/3)) {
        for (var i = gap; i < len; i++) {
            temp = arr[i];
            for (var j = i - gap; j >= 0 && arr[j] > temp; j -= gap) {
                arr[j + gap] = arr[j];
            }
            arr[j + gap] = temp;
        }
    }
    return arr;
}

var arr = new Array(7,53,24,45,21,5,78,43,4,25,43);
var rt = shellSort(arr);

console.log(rt);

File: shell/shell.go
package main

func main() {
	var arr = []int{7,53,24,45,21,5,78,43,4,25,43}
	shellsort(arr)
	for i := 0; i < len(arr); i++ {
		println(arr[i])
	}
}

func shellsort(arr []int) []int {
	length := len(arr)
	gap := 1
	for gap < gap/3 {
		gap = gap * 3 + 1
	}
	for gap > 0 {
		for i := gap; i < length; i++ {
			temp := arr[i]
			j := i - gap
			for j >= 0 && arr[j] > temp {
				arr[j + gap] = arr[j]
				j -= gap
			}
			arr[j + gap] = temp
		}
		gap = gap / 3
	}
	return arr
}